src/test/kotlin/org/jmailen/gradle/kotlinter/functional/WithGradleTest.kt

```diff
@@ -46,5 +46,5 @@ abstract class WithGradleTest {
 private fun WithGradleTest.defaultRunner(vararg args: String) =
     GradleRunner.create()
         .withProjectDir(testProjectDir.root)
-        .withArguments(args.toList() + "--stacktrace")
+        .withArguments(args.toList() + listOf("--stacktrace", "--configuration-cache"))
```

Comment:
🔧 
Can you elaborate why it is needed? At first glance it doesn't sound right - every _first_ call will result in cache miss, right? So in almost every test we'll see _"Calculating task graph as no configuration cache is available for tasks:"_ warning, which means it doesn't really change anything. If there is a test that does actually leverage it - I'd personally suggest being more explicit and adding explicit arguments there. 
This feels like checking if "_first_ kotlinter task invocation does not break the build", not if "git hook tasks work with configuration cache". 
I might be missing something here, I haven't ever touched git hook tasks, and I don't really use them so my input might not be valid 🤷

Reply:
The reason is not that the tests themselves make use of the configuration cache.

The motiviation for this change is that this will let the tests fail if any of the kolinter task is conflicting with the configuration cache, for example like the hook, which was accessing the project at configuration time.